### ISSUE TITLE
Do not require a Wasm VM on the light client by fetching Babe config through other means

### ISSUE THREAD

**Issue by tomaka** [paritytech/substrate#6952]
At the moment, we fetch the Babe configuration by calling the `Babe_configuration` runtime function on the genesis block.
Since this is called on the genesis block, the returned value is the same every single time and is fully predictable from the chain specs alone.

The light client requires this Babe configuration in order to sync, and therefore needs to implement a Wasm virtual machine.

While the `Babe_configuration` runtime function is a convenient way of providing the Babe configuration, it is also the one and only runtime call that the light client ever needs to make.

It feels like a waste to have to ship light clients with a fully functional Wasm virtual machine just for one specific configuration that is always known ahead of time.

cc @andresilva

**Comment by tomaka** [comment]
> It feels like a waste to have to ship light clients with a fully functional Wasm virtual machine just for one specific configuration that is always known ahead of time.

I've now realized that this is actually not true.
Obtaining the metadata requires calling a runtime function (`Metadata_metadata`), and I don't think there's a secure way to obtain this metadata that doesn't involve having a Wasm interpreter.

**Comment by bkchr** [comment]
Why is metadata the important function here? Because the UI on top of the light client will need to know the metadata to know how the runtime looks like?

**Comment by tomaka** [comment]
Yeah. In particular, as far as I understand, it's not possible to create transactions without the metadata. And a light client that doesn't provide all the info required to create transactions wouldn't be very useful.

**Comment by bkchr** [comment]
Yeah that is right.
